dynamodb/src/main/scala/zio/dynamodb/DynamoDBExecutorImpl.scala

```diff
@@ -215,15 +218,14 @@ private[dynamodb] final case class DynamoDBExecutorImpl private (dynamoDb: Dynam
         acc ++ ((TableName(tableName), l.flatMap(f)))
     }
 
-  private def doBatchWriteItem(batchWriteItem: BatchWriteItem): ZIO[Clock, Throwable, BatchWriteItem.Response] =
+  // TODO: Should be tail recursive
```

Comment:
I think having no tail recursion is OK here as the function returns an effect which makes it stack safe